contact-book-python-textual/source_code_step_2/rpcontacts/tui.py

```diff
@@ -0,0 +1,71 @@
+from textual.app import App
+from textual.containers import Grid, Horizontal, Vertical
+from textual.screen import Screen
+from textual.widgets import Button, DataTable, Footer, Header, Label, Static
+
+
+class ContactsApp(App):
+    CSS_PATH = "rpcontacts.tcss"
+    BINDINGS = [
+        ("m", "toggle_dark", "Toggle dark mode"),
+        ("a", "add", "Add"),
+        ("d", "delete", "Delete"),
+        ("c", "clear", "Clear All"),
```

Comment:
Later in the tutorial you'll name the method `.action_clear_all()`. You can either decide to rename the binding here, or rename the method later:

```suggestion
        ("c", "clear_all", "Clear All"),
```